clusterDeployer.py

```diff
@@ -29,222 +29,654 @@
-        logger.error_and_exit(str(ret))
-
-    # Fix cases where virsh net-start fails with error "... interface virbr0: File exists"
-    cmd = "ip link delete virbr0"
-    h.run(cmd)  # ignore return code - it might fail if virbr did not exist
-
-    cmd = f"virsh net-define {bridge_xml}"
-    h.run_or_die(cmd)
-
-    # set interface down before starting bridge as otherwise bridge start might fail if interface
-    # already got an IP address in same network as bridge
-    h.run(f"ip link set {api_network} down")
-
-    cmd = "virsh net-start default"
-    h.run_or_die(cmd)
-
-    h.run(f"ip link set {api_network} up")
-
-
-def limit_dhcp_range(h: host.Host, old_range: str, new_range: str) -> None:
-    # restrict dynamic dhcp range: we use static dhcp ip addresses; however, those addresses might have been used
-    # through the dynamic dhcp by any systems such as systems ready to be installed.
-    cmd = "virsh net-dumpxml default"
-    ret = h.run(cmd)
-    if f"range start='{old_range}'" in ret.out:
-        host_xml = f"<range start='{old_range}' end='192.168.122.254'/>"
-        cmd = f"virsh net-update default delete ip-dhcp-range \"{host_xml}\" --live --config"
-        r = h.run(cmd)
-        logger.debug(r.err if r.err else r.out)
-
-        host_xml = f"<range start='{new_range}' end='192.168.122.254'/>"
-        cmd = f"virsh net-update default add ip-dhcp-range \"{host_xml}\" --live --config"
-        r = h.run(cmd)
-        logger.debug(r.err if r.err else r.out)
-
-
-def network_xml(ip: str, dhcp_range: Optional[Tuple[str, str]] = None) -> str:
-    if dhcp_range is None:
-        dhcp_part = ""
-    else:
-        dhcp_part = f"""<dhcp>
-  <range start='{dhcp_range[0]}' end='{dhcp_range[1]}'/>
-  </dhcp>"""
-
-    return f"""
-<network>
-  <name>default</name>
-  <forward mode='nat'/>
-  <bridge name='virbr0' stp='off' delay='0'/>
-  <ip address='{ip}' netmask='255.255.0.0'>
-    {dhcp_part}
-  </ip>
-</network>
-  """
-
-
-def configure_bridge(h: host.Host, api_network: str) -> None:
-    hostname = h.hostname()
-    cmd = "systemctl enable libvirtd"
-    h.run_or_die(cmd)
-    cmd = "systemctl start libvirtd"
-    h.run_or_die(cmd)
-
-    # stp must be disabled or it might conflict with default configuration of some physical switches
-    # 'bridge' section of network 'default' can't be updated => destroy and recreate
-    # check that default exists and contains stp=off
-    cmd = "virsh net-dumpxml default"
-    ret = h.run(cmd)
-
-    if "stp='off'" not in ret.out:
-        logger.info("Destoying and recreating bridge")
-        logger.info(f"creating default-net.xml on {hostname}")
-        if hostname == "localhost":
-            contents = network_xml('192.168.122.1', ('192.168.122.129', '192.168.122.254'))
+    hostconn: host.Host
+    bridge: VirBridge
+    config: HostConfig
+    k8s_master_nodes: List[ClusterNode]
+    k8s_worker_nodes: List[ClusterNode]
+    k8s_nodes: List[ClusterNode]
```

Comment:
can we have a function instead of cashing `k8s_master_nodes + k8s_worker_nodes`

Reply:
Sure.